### ISSUE TITLE
After installed the drivers graphic ms-dos application shows problems 

### ISSUE THREAD

**Issue by 1985a** [schellingb/dosbox-pure#281]
As the title says, I've installed the drives for 3DFX, and any DOS applications got kinda freeze the system itselft, but if you press the famous combination ctrl+alt+del, the blue screen come up, then press the key enter, now you can see the application working, but not at the normal way.

**Comment by Winchester95** [comment]
Can you provide windows 98se link?

**Comment by 1985a** [comment]
> You should list the exact steps needed to recreate your problem.
> 
> But for me i've installed Windows 98 SE and used the last officially released [Voodoo 1 drivers](https://3dfxarchive.com/voodoo1.htm) (v3.01.00) and the last [DirectX drivers](http://falconfly.3dfx.pl/directx.htm) (Dec 2006) and i haven't had any problems with DOS. Although with Win 98 SE it already comes with the same Voodoo driver, so it's not even needed unless you want the 3dfx Voodoo settings panel.

I did the thing you said here. Uninstalled the driver and went to use the Microsoft instead. The problem persists. This Saturday I'll play with another installation to see what can I do.

BTW, the steps that I performed were installed the system, install a few games, tested it, then trying to install the drivers, loaded the driver and did install it. Restart the system, and from here, ms-dos was facing that problem. The system was using 16bit color.

In order to see the MS-DOS window, I need to press the combination keys, as I said before.


Any mistake, English is a language that I'm learning.
Here two screen capture
https://imgur.com/a/mufdK8b

**Comment by PoloniumRain** [comment]
> Can you provide windows 98se link?

[Here.](https://archive.org/details/win98se_201801)

> BTW, the steps that I performed were installed the system, install a few games, tested it, then trying to install the drivers, loaded the driver and did install it. Restart the system, and from here, ms-dos was facing that problem. The system was using 16bit color.

I didn't understand what you meant at first, but you're right, the MS-DOS Prompt within Windows 98 SE will freeze everything when it's opened. But it's not related to the Voodoo drivers. It still freezes with a clean install of Windows 98 SE, with no drivers installed. 

It has something to do with the emulated CPU speed (core Options > Performance Options > Emulated Performance). If it's set to AUTO it will almost entirely freeze up and the mouse cursor gets stuck. But if a much slower CPU is emulated like a 486DX 33MHz, then the cursor can now be moved when the MS-DOS Prompt is open, but it's still laggy. I highly doubt this is a Windows 98 thing because i've never seen this happen with real hardware or when using another emulator like PCem. Some software and games also use the MS-DOS-Prompt for installation or extracting files, so they either fail to install or take ages. Hopefully @schellingb can take a look at this.

**Comment by schellingb** [comment]
Sorry I had to remove the link. You're on your own with that stuff.

Regarding the MS-DOS console, it might seem funny, but emulating DOS inside Windows inside DOSBox is a very complex problem. DOS inside windows is some form of "virtual machine" which means the emulated hardware needs to be quite accurate to make this work correctly. DOSBox (and DOSBox Pure) might just not be accurate enough for that. We are targeting games so this has somewhat of a lower priority.

Is there a game that needs the DOS console? Could this be worked around by using selecting "Restart in MS-DOS mode" from the Shutdown menu?

**Comment by 1985a** [comment]
> 
> Is there a game that needs the DOS console? Could this be worked around by using selecting "Restart in MS-DOS mode" from the Shutdown menu?


There are a few of them that need a DOS console in order to load the game, like Duke Nuke 3D. I'm emulating a Sound Blaster 16, and this game needs to know if I have installed that sound card. As I said before, by pressing the combination, I can do the thing that I wanted to do. So for me, this issues is not high priority.

I just felt the need to share this information.

Mmm, I didn't test Restart in MS-DOS yet.

**Comment by PoloniumRain** [comment]
> Is there a game that needs the DOS console? 

Yeah but i can't remember which game it was now. Software like the [Unofficial Service Pack 3](http://www.htasoft.com/u98sesp/) for Win 98 will freeze roughly 70% of the time when it opens the MS-DOS Prompt. The other 30% it will either work ok or have some lag. Freezing seems completely random and can't be replicated (i've been trying for ages). But strangely, freezing is slightly reduced after USP3 has been installed, at least when running USP3 again afterwards. Setting a fixed CPU speed before booting Windows will also slightly reduce freezing. The CTRL+ALT+Delete method already mentioned doesn't always work either, depends on the program/game and randomness. One time when DOS froze it actually crashed RA but that's only happened once in 50+ times.

> Could this be worked around by using selecting "Restart in MS-DOS mode"

I've always managed to get things working with enough attempts, so haven't tried this yet.

> Sorry I had to remove the link.

Ah sorry about that! Forgot that link had a key lol.

**Comment by schellingb** [comment]
Can you try to toggle the core option `System > Advanced > Force Normal Core in OS` to ON while doing console stuff? The 'Normal Core' is slower but should be more compatible/stable with stuff that requires more accurate CPU emulation.

**Comment by PoloniumRain** [comment]
Normal Core helps a lot but has problems...

Positive: 
The MS-DOS Prompt will never freeze when opened.
Negative:
It can't be used for input, because now it's just a black screen with no visible text:

![MS-DOS](https://user-images.githubusercontent.com/50634894/175788873-495942a8-50e1-4874-af08-73517163ae97.png)

Positive: 
All games/software will now install ok (so far at least).
Negative:
Apart from performance absolutely tanking with Normal Core, some things also wont work anymore (3DMark 99 will completely lock up the OS on the second game test).

So Normal Core is good to use for any software/games that run the MS-DOS Prompt during installation, but do not require any input. And none of the stuff i've tried has needed input so far so it's a good workaround for this specific issue at least.
